### PR TITLE
#833 providers json text to numeric

### PR DESCRIPTION
**Pull request by vancamti**
waiting for reponse on https://github.com/onroerenderfgoed/atramhasis/issues/833#issuecomment-1550863940

### CODE REVIEW

File: atramhasis/scripts/migrate_sqlalchemy_providers.py [hunk 1]
```diff
@@ -187,8 +187,9 @@ def migrate(skos_registry: Registry, session: Session):
             )
         else:
             provider.metadata['atramhasis.id_generation_strategy'] = 'NUMERIC'
-        db_provider.meta = json.loads(json.dumps(provider.metadata, default=json_serial))
-
+        meta = json.loads(json.dumps(provider.metadata, default=json_serial))
+        meta["id"] = provider.conceptscheme_id
```
Comment (koenedaele) [inline atramhasis/scripts/migrate_sqlalchemy_providers.py:191]: This makes the problem worse as it now fully overwrites the old provider id with the conceptscheme_id. We want the old provider ID to remain functional.

File: tests/scripts/test_migratie_sqlalchemy_providers.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def test_migrate(self):
         self.assertEqual('urn:x-skosprovider:%s:%s', provider.uri_pattern)
         self.assertEqual(
             {
-                'id': 'EXTRA',
+                'id': int(provider.id),
```
Comment (koenedaele) [inline tests/scripts/test_migratie_sqlalchemy_providers.py:65]: This should actually be 'EXTRA', so the old test was correct.

**Review by koenedaele — CHANGES_REQUESTED** [review]
We want the old ID's to remain fully functional since they are used in document URL's. Solution is probably somewhere in the code that registers the providers with the registry.

**Comment by koenedaele** [comment]
> Do we want both http://localhost:6543/conceptschemes/HERITAGETYPE and http://localhost:6543/conceptschemes/6 to work or should only http://localhost:6543/conceptschemes/HERITAGETYPE work ? Is it required that the provider gets a numerical id `db_provider.id = provider.conceptscheme_id` or can it just be `db_provider.id = db_provider.id = provider.get_vocabulary_id()` ? (which makes the old url fully functional)

The numeric id does not need to be exposed through the URL. So `db_provider.id  = provider.get_vocabulary_id()`

**Review by Wim-De-Clercq — APPROVED** [review]
I do believe that is correct... suspiciously small fix :raised_eyebrow:

**Review by koenedaele — APPROVED** [review]

**Pull request merged**